pkg/provider/cloud/azure/provider.go

```diff
@@ -74,37 +75,79 @@ func New(dc *kubermaticv1.Datacenter, secretKeyGetter provider.SecretKeySelector
-	"southeastasia":      2,
-	"centralus":          3,
-	"eastus":             3,
-	"eastus2":            3,
-	"westus":             3,
-	"northcentralus":     3,
-	"southcentralus":     3,
-	"northeurope":        3,
-	"westeurope":         3,
-	"japanwest":          2,
-	"japaneast":          2,
-	"brazilsouth":        2,
-	"australiaeast":      2,
-	"australiasoutheast": 2,
-	"southindia":         2,
-	"centralindia":       2,
-	"westindia":          2,
-	"canadacentral":      3,
-	"canadaeast":         2,
-	"uksouth":            2,
-	"ukwest":             2,
-	"westcentralus":      2,
-	"westus2":            2,
-	"koreacentral":       2,
-	"koreasouth":         2,
+func getRegionFaultDomainCount(location string) (int32, error) {
+	// Azure API doesn't allow programmatically getting the number of available fault domains in a given region.
+	// We must therefore hardcode these based on https://docs.microsoft.com/en-us/azure/virtual-machines/windows/manage-availability
+	//
+	// The list of region codes was generated by `az vm list-skus --resource-type availabilitySets --query '[?name==`Aligned`].{Location:locationInfo[0].location, MaximumFaultDomainCount:capabilities[0].value}'`.
+	var faultDomainsPerRegion = map[string]int32{
+		"AustraliaCentral":   2,
+		"AustraliaCentral2":  2,
+		"australiaeast":      2,
+		"australiasoutheast": 2,
+		"brazilsouth":        3,
+		"BrazilSoutheast":    2,
+		"CanadaCentral":      3,
+		"CanadaEast":         2,
+		"CentralIndia":       3,
+		"centralus":          3,
+		"CentralUSEUAP":      1,
+		"eastasia":           2,
+		"eastus":             3,
+		"eastus2":            3,
+		"EastUS2EUAP":        2,
+		"EastUSSTG":          1,
+		"FranceCentral":      3,
+		"FranceSouth":        2,
+		"GermanyNorth":       2,
+		"GermanyWestCentral": 2,
+		"IsraelCentral":      2,
+		"ItalyNorth":         2,
+		"japaneast":          3,
+		"japanwest":          2,
+		"JioIndiaCentral":    2,
+		"JioIndiaWest":       2,
+		"KoreaCentral":       2,
+		"KoreaSouth":         2,
+		"MalaysiaSouth":      2,
+		"MexicoCentral":      2,
+		"NewZealandNorth":    1,
+		"northcentralus":     3,
+		"northeurope":        3,
+		"NorwayEast":         2,
+		"NorwayWest":         2,
+		"PolandCentral":      2,
+		"QatarCentral":       2,
+		"SouthAfricaNorth":   2,
+		"SouthAfricaWest":    2,
+		"southcentralus":     3,
+		"SouthCentralUSSTG":  2,
+		"southeastasia":      2,
+		"SouthIndia":         2,
+		"SpainCentral":       2,
+		"SwedenCentral":      3,
+		"SwedenSouth":        2,
+		"SwitzerlandNorth":   2,
+		"SwitzerlandWest":    2,
+		"TaiwanNorth":        2,
+		"TaiwanNorthwest":    2,
+		"UAECentral":         2,
+		"UAENorth":           2,
+		"uksouth":            2,
+		"ukwest":             2,
+		"westcentralus":      2,
+		"westeurope":         3,
+		"WestIndia":          2,
+		"westus":             3,
+		"westus2":            3,
+		"WestUS3":            3,
+	}
+	for region := range faultDomainsPerRegion {
```

Comment:
You could simply `for region, count := range ...` and then avoid the 2nd slice access in line 147, no? It's fine for the loop to overwrite `count` on every iteration.